sysvar: add variables in 6.4 doc (#11229)
pingcap/docs: system-variables.md

- Default value: `OFF`
- This variable is used to set whether the `AUTO_INCREMENT` property of a column is allowed to be removed by executing `ALTER TABLE MODIFY` or `ALTER TABLE CHANGE` statements. It is not allowed by default.

> ### tidb_analyze_partition_concurrency

Reviewer: ```suggestion
### tidb_analyze_partition_concurrency <span class="version-mark">New in v6.4.0</span>
```

Author: I don't think it's a good idea to say it's new in v6.4.0 although it is a fact. The feature owner wants to make it public in a future release. If we say it's new in v6.4.0, it will sound like the related unreleased feature is new in v6.4.0.

> **Warning:**
>
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes

> - Default value: `1`

Reviewer 2: Is there a range for this variable?

Author: We do not recommend users to set it before the related feature is released. So the value range is not necessary IMO.

Reviewer: From the code in `sessionctx/variable/sysvar.go`:
```go
        {Scope: ScopeGlobal | ScopeSession, Name: TiDBAnalyzePartitionConcurrency, Value: strconv.FormatInt(DefTiDBAnalyzePartitionConcurrency, 10),
                MinValue: 1, MaxValue: uint64(config.GetGlobalConfig().Performance.AnalyzePartitionConcurrencyQuota), SetSession: func(s *SessionVars, val string) error {
                        s.AnalyzePartitionConcurrency = int(TidbOptInt64(val, DefTiDBAnalyzePartitionConcurrency))
                        return nil
                }},
```

And in `config/config.go`:
```go
        AnalyzePartitionConcurrencyQuota uint   `toml:"analyze-partition-concurrency-quota" json:"analyze-partition-concurrency-quota"`
...
                AnalyzePartitionConcurrencyQuota: 16,
```

So looks like the max is set to the value of `analyze-partition-concurrency-quota` which seems to be 16 by default.

I would be ok to leave this out of the docs for the reason that @TomShawn just said.

Reviewer: @TomShawn I assume the `analyze-partition-concurrency-quota` config is added to the docs in another PR?

Author: Yes. I'll ask the related developer to provide the doc input for `analyze-partition-concurrency-quota` in another PR.

- When this variable is set to `ON`, you can view visual execution plans in TiDB Dashboard. Note that TiDB Dashboard only provides visual display for execution plans generated after this variable is enabled.
- You can execute the `SELECT tidb_decode_binary_plan('xxx...')` statement to parse the specific plan from a binary plan.

### tidb_gogc_tuner_threshold <span class="version-mark">New in v6.4.0</span>

- Scope: GLOBAL
- Persists to cluster:  No, only applicable to the current TiDB instance that you are connecting to.
- Default value: `0.6`
- Range: `[0, 0.9)`

> - This variable specifies the maximum memory usage for GOGC Tuner. When the memory usage of GOGC Tuner exceeds this threshold, GOGC Tuner stops working.

Author: ```suggestion
- This variable specifies the maximum memory threshold for tuning GOGC. When the memory exceeds this threshold, GOGC Tuner stops working.
```

- This variable sets the concurrency of the `MergeJoin` operator when a query is executed.
- It is **NOT recommended** to set this variable. Modifying the value of this variable might cause data correctness issues.

> ### tidb_merge_partition_stats_concurrency

Reviewer: ```suggestion
### tidb_merge_partition_stats_concurrency <span class="version-mark">New in v6.4.0</span>
```
